### PR TITLE
Polling API update

### PR DESCRIPTION
**Pull request by g2vinay**

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java [hunk 1]
```diff
@@ -138,6 +139,47 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
         this.autoPollingEnabled = true;
     }
 
+    /**
+     * Create a {@link Poller} instance with poll interval and poll operation. The polling starts immediately by invoking {@code pollOperation}.
+     * The next poll cycle will be defined by {@code retryAfter} value in {@link PollResponse}.
+     * In absence of {@code retryAfter}, the {@link Poller} will use {@code pollInterval}.
+     *
+     * <p><strong>Code Sample - Create poller object</strong></p>
+     * {@codesnippet com.azure.core.util.polling.poller.initialize.interval.polloperation}
```
Comment (JonathanGiles) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java]: Need a new code sample

Comment (g2vinay) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java]: yeah, just want to make sure, API surface change is good to go, before putting effort for Tests + Samples.

**Comment by g2vinay** [comment]
> I'm fine with the general concept. I would like to see the constructors flow through to each other, rather than duplicate their impl.

Sounds, Good.
Yeah the refactor is simple for constructors.
Main constructor will be 
pollInterval, pollOperation, cancelOperation, activationOperation

Then other permutations calling main constructor will be:
pollInterval, pollOperation
pollInterval, pollOperation, cancelOperation
pollInterval, pollOperation, activationOperation

File: sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java [hunk 2]
```diff
@@ -150,14 +153,54 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
...
+     * @param pollOperation The polling operation to be called by the {@link Poller} instance. This is a callback into the client library,
+     * which must never return {@code null}, and which must always have a non-null {@link OperationStatus}.
+     * {@link Mono} returned from poll operation should never return {@link Mono#error(Throwable)}.If any unexpected scenario happens in poll operation,
+     * it should handle it and return a valid {@link PollResponse}. However if poll operation returns {@link Mono#error(Throwable)},
+     * the {@link Poller} will disregard that and continue to poll.
+     * @param activationOperation The activation operation to be called by the {@link Poller} instance before calling {@code pollOperation}. It can
+     * be {@code null} which will indicate to the {@link Poller} that {@code pollOperation} can be called straight away.
+     * @throws IllegalArgumentException if {@code pollInterval} is less than or equal to zero and if {@code pollInterval} or {@code pollOperation} are {@code null}
+     */
+    public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation, Supplier<Mono<T>> activationOperation) {
```
Comment (JonathanGiles) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java]: Which of these constructors do you need vs which did you add for completeness? Let's keep the number added to the minimum.

Comment (g2vinay) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java]: This one was added for completeness,   (pollInterval, pollOperation, activationOperation)
Removing it now then,

**Pull request merged**